### PR TITLE
Add styles for TeacherDashboard page. Add getTeacherCourses method.

### PR DESCRIPTION
**Pull request by OGoncharenko**
Add styles for TeacherDashboard page. Add getTeacherCourses method.

### CODE REVIEW

**Comment by OGoncharenko** [comment]
Teacher dashboard view
image

File: frontend/src/components/layout/nav.module.css
Comment (lauradurandv) [inline frontend/src/components/layout/nav.module.css:1]: The styling here is tricky because this pr is 1 merge behind (Laura's: complete front page), and when I merged, I was 1 merge behind as well (John's: Sign up page). As a result, when I merged, I added class names for nav styling which is now on the main branch but not visible here.

File: frontend/src/components/layout/Nav.jsx
Comment (lauradurandv) [inline frontend/src/components/layout/Nav.jsx:1]: Going with the styling that is currently on the main branch for the **_nav.module.css_** file, I would say it would be easiest to make the following changes: 

In the return statement, 
1. The second **div** should have a className={styles.listContainer}
2. The **ul** should have className={styles.list}
3. The **li** should have the className={styles.listItem}

I do want to mention that I am not sure this is the best course of action to take, but this solution makes the most sense to me. Because this pr is one merge behind, I imagine doing the code above would help reduce merge conflicts. I'd appreciate any of our mentor's advice on this.

File: frontend/src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (lauradurandv) [inline frontend/src/pages/TeacherDashboard/TeacherDashboard.jsx:1]: In the most recent merged pull request to the readme file, I added directions for adding a similar environment variable. I dont know if your base url is the same as the one in the readme, but if so, I think for the sake of consistency (if it matters) we should name it the same. Although it is different, I would also add this env var to the readme file.

File: frontend/src/components/layout/nav.module.css
Comment (OGoncharenko) [reply frontend/src/components/layout/nav.module.css:1]: The changes in styles for nav were made by me, that helped to me to keep working on my pages. I will accept your changes in merge conflicts.

File: frontend/src/components/layout/Nav.jsx
Comment (OGoncharenko) [reply frontend/src/components/layout/Nav.jsx:1]: I will accept your changes in your nav

File: frontend/src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (OGoncharenko) [reply frontend/src/pages/TeacherDashboard/TeacherDashboard.jsx:1]: For the URL we have different variable name

File: src/App.jsx
Comment (ElijahMSmith) [inline src/App.jsx:18]: I am of the opinion that we should have one dashboard. If you are a teacher, you should see the courses you manage. If you are a student, you should see the courses you are currently taking, and this component would be at `/dashboard`.

Since this seems to be specifically focused on the teacher side of things, maybe we can leave the student part out until we have a proper login/signup flow and for now we just rename this component.

File: src/pages/Courses/Courses.jsx
Comment (ElijahMSmith) [inline src/pages/Courses/Courses.jsx:1]: It's not clear to me why we need this component. It currently doesn't do anything and isn't styled. I think we should just stick to the dashboard and remove this, but I will leave this up to you if there is more planned here.

File: src/index.css
Comment (ElijahMSmith) [inline src/index.css:15]: Let's delete this rather that just a comment, which might make people think it should be added back in.

File: src/components/SideBar/SideBarStyles.module.css
Comment (ElijahMSmith) [inline src/components/SideBar/SideBarStyles.module.css:1]: I think we should keep our CSS modules under one name schema: `<component name>.module.css`. Let's not change this here though, we should do this in a separate commit after this one.

File: src/components/SideBarItems/SideBarItems.jsx
Comment (ElijahMSmith) [inline src/components/SideBarItems/SideBarItems.jsx:8]: We can put this outside and above this component. If it is a constant, we don't need to re-create this every time the component re-renders.

File: src/components/SideBarItems/SideBarItems.jsx
Comment (ElijahMSmith) [inline src/components/SideBarItems/SideBarItems.jsx:1]: We don't need to import React in modern versions of React. I get a red squiggle every time we do this because the `eslint` plugin is reminding me it's not necessary. Do you have the plugin installed and running?

File: src/index.css
Comment (ElijahMSmith) [reply src/index.css:15]: Same for above with the font property.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [inline src/pages/TeacherDashboard/TeacherDashboard.jsx:35]: Let's `console.error` for errors.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [inline src/pages/TeacherDashboard/TeacherDashboard.jsx:44]: We should not just return null here, which will make our page go blank. In theory, the useEffect should navigate us back if the token is not valid, but we should still render a valid page until that happens. The courses array should be empty so your layout might work anyways and just not show any course content, which is fine if so.

File: src/App.jsx
Comment (ElijahMSmith) [reply src/App.jsx:18]: As a further example, it wouldn't make much sense for me to have a route `/teacher/dashboard` but no route `/teacher`. I would expect to only have a child route if the parent exists and it logically makes sense.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [inline src/pages/TeacherDashboard/TeacherDashboard.jsx:51]: Nit: Maybe "Welcome to your dashboard!"

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [inline src/pages/TeacherDashboard/TeacherDashboard.jsx:67]: Nit: "You don't have any courses yet. To get started, add a new course!"

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [inline src/pages/TeacherDashboard/TeacherDashboard.jsx:85]: This doesn't need to be in a div if it's only one item.

File: server/controllers/courseController.mjs
Comment (ElijahMSmith) [inline server/controllers/courseController.mjs:63]: This comment can be removed since it doesn't add any useful information.

File: server/routes/courseRoutes.mjs
Comment (ElijahMSmith) [inline server/routes/courseRoutes.mjs:30]: This should be `/api/course/teacher_courses`

File: server/routes/courseRoutes.mjs
Comment (ElijahMSmith) [inline server/routes/courseRoutes.mjs:44]: I don't think we should have this route. We should already have a route that can get a list of courses, and we should modify that to make it work.

Maybe we could modify the `/user/:userId/course` route in `userCourseRoutes.mjs` to return the courses a user is TAKING if they are a student or the courses the user OWNS if they are a teacher?

**Review by ElijahMSmith — CHANGES_REQUESTED** [review]
Lots of good things here! Left some small cleanups and a few more high-level discussion points. Ping me again when you are finished making changes!

File: src/pages/Courses/Courses.jsx
Comment (OGoncharenko) [reply src/pages/Courses/Courses.jsx:1]: The idea of this component to let the Teacher see all the courses that we have in our app. It supposed to be a feature to assign different courses to the class of students that the teacher has. For MVP we decided to postpone the feature with Classe and Students, but I believe that the teacher should still see all the courses.

File: server/routes/courseRoutes.mjs
Comment (OGoncharenko) [reply server/routes/courseRoutes.mjs:44]: Maybe we can leave this as-is and refactor it in the future?

File: src/App.jsx
Comment (OGoncharenko) [reply src/App.jsx:18]: I agree! I can make those changes in the next PR. I think we can merge this one. We can refactor it how to render it when working Sign Up and Sign In components will be merged, and User Dashboard also.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (OGoncharenko) [reply src/pages/TeacherDashboard/TeacherDashboard.jsx:85]: I prefer to leave it for easy styling

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [inline src/pages/TeacherDashboard/TeacherDashboard.jsx:7]: I've just merged in a change that changes Laura's code to use VITE_API_BASE_URL, so let's stick with that instead of VITE_BASE_URL and be consistent everywhere. Apologies if I asked you to change it initially.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [inline src/pages/TeacherDashboard/TeacherDashboard.jsx:14]: When you rebase the new changes, this will need to be '/' again since I moved the FrontPage component to that route.

File: frontend/src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [reply frontend/src/pages/TeacherDashboard/TeacherDashboard.jsx:1]: See other comment - I've changed the URL to VITE_API_BASE_URL everywhere in the most recent set of commits, let's stick with that.

File: server/routes/courseRoutes.mjs
Comment (ElijahMSmith) [reply server/routes/courseRoutes.mjs:44]: If we're adding this new API route in this commit, why not spend a little extra time and get things right the first time?

Generally, once we release an API route, it is difficult to make big changes (like change the route name, changing the schema of the request/response, etc.) because it will break applications that are using that route. However, given that we are on a deadline and we don't have clients using our API other than the frontend, I will give this a pass this time if it's going to slow you down too much.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (ElijahMSmith) [reply src/pages/TeacherDashboard/TeacherDashboard.jsx:83]: I see that you are using this to center the button in the flex container. You could also give `alignItems: center` to the container to accomplish the same thing.

**Review by ElijahMSmith — APPROVED** [review]
LGTM with me once you rebase and change the URL environment variable and front page route, as per comments.

File: src/pages/TeacherDashboard/TeacherDashboard.jsx
Comment (OGoncharenko) [reply src/pages/TeacherDashboard/TeacherDashboard.jsx:83]: Didn't work for the table view

**Pull request merged**
